overlaysFromChannelsExporter.nix

```diff
@@ -31,14 +33,24 @@ let
 
       **/
     let
-      inherit (builtins) mapAttrs attrNames concatMap listToAttrs;
+      inherit (builtins) mapAttrs foldl' filter head attrNames attrValues concatMap listToAttrs elem;
       nameValuePair = name: value: { inherit name value; };
 
+      # just pull out one arch from the system-spaced pkgs to get access to channels
+      # overlays can be safely evaluated on any arch
+      channels = head (attrValues pkgs);
+
       pathStr = path: builtins.concatStringsSep "/" path;
 
       channelNames = attrNames channels;
       overlayNames = overlay: attrNames (overlay null null);
 
+      /* workaround to detect overlays owned by this flake, by filtering out overlays from others */
+      # get all overlays from inputs
```

Comment:
Whats up with different type of comments :D

Reply:
I was going for a header, but I think I'm going to move that comment to the top next to the `inputs` argument description. So its clear why its needed.